File: 2021/1/elixir/aoc_2021_1.exs
defmodule AOC do
  def read_input() do
    File.stream!("../input")
    |> Stream.map(fn (line) -> line |> Integer.parse |> elem(0) end)
    |> Enum.to_list
  end

  def run_part_a(values) do
    1..(length(values) - 1)
    |> Enum.filter(fn index -> Enum.at(values, index) > Enum.at(values, index - 1) end)
    |> length
  end

  def run_part_b(values) do
    Enum.chunk_every(values, 3, 1, :discard)
    |> Enum.map(fn chunk -> chunk |> Enum.sum end)
    |> run_part_a
  end
end

IO.puts("Part A: #{AOC.read_input() |> AOC.run_part_a}")
IO.puts("Part B: #{AOC.read_input() |> AOC.run_part_b}")

File: 2021/1/python/aoc_2021_1.py

def run_part_a(values):
    return len([values[i] for i in range(len(values)) if i > 0 and values[i] > values[i - 1]])


def run_part_b(values):
    return run_part_a([sum(values[i:i+3]) for i in range(len(values) - 2)])


if __name__ == '__main__':
    with open('../input') as input_file:
        input_values = [int(line) for line in input_file.readlines() if line]
        print(f'Part A: {run_part_a(input_values)}\nPart B: {run_part_b(input_values)}')
